kube-app-testing.sh

```diff
@@ -584,42 +533,8 @@ create_gs_cluster () {
     sleep 30
   done
 
-  info "Creating key-pair for tenant cluster access."
-  # create a key pair (must be stored directly in a variable)
-  _key_pair=$(curl ${GS_API_URL}/v4/clusters/${CLUSTER_ID}/key-pairs/ -X POST \
-    -H "Content-Type: application/json" \
-    -H "Authorization: giantswarm ${GSAPI_AUTH_TOKEN}" \
-    -d "$(gen_gs_blob keypair)")
-
-  # check that we actually got a key pair back
-  grep -q "certificate_authority_data" <<< $_key_pair
-  if [[ "$?" -gt 0 ]]; then
-    err "Key pair creation failed."
-    exit 3
-  fi
-
-  # parse required fields from key pair creation response in order to
-  # create a kubeconfig for the tenant cluster
-  TC_API_URI=$(curl -s -H "Authorization: giantswarm ${GSAPI_AUTH_TOKEN}" ${GS_API_URL}/v5/clusters/${CLUSTER_ID}/ | jq -r .api_endpoint)
-  CA_CERT=$(echo $_key_pair | jq -r '.certificate_authority_data | @base64')
-  CLIENT_CERT=$(echo $_key_pair | jq -r '.client_certificate_data | @base64')
-  CLIENT_KEY=$(echo $_key_pair | jq -r '.client_key_data | @base64')
-
-  # align config with kind clusters
-  if [[ ! -d ${CONFIG_DIR} ]]; then
-    mkdir ${CONFIG_DIR}
-  fi
-
-  info "Templating kubeconfig out to ${KUBECONFIG}"
-  # create the kubeconfig
-  gen_gs_blob kubeconfig ${KUBECONFIG}
-
-  # update Security Group to allow access
-  update_aws_sec_group ${CLUSTER_ID}
-
-  # make sure the ingress rule has taken effect before we attempt to connect
-  info "Sleeping for 30 seconds to ensure ingress rule has been applied."
-  sleep 30
```

Comment:
It's probably better to leave this in (maybe reduce the sleep time). It's to allow the AWS security group time to update before we attempt to talk to the GS cluster API.